### PR TITLE
Geocoding readme improvements

### PR DESCRIPTION
**Pull request by tTakeda09**
**Sample**

Geocoding

**Summary**

- Introduce How to use  the sample section
- Make the How it works section only with what's behind in order to show the function 

**Tests**

Read

**ArcGIS Maps SDK Version**

### CODE REVIEW

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+**Note:** This sample is only set up to work with a mouse and keyboard.
+
+## How it works
 
+1. Have a map with the mesh colliders enabled in the level. The mesh collider is required for using raycast. 
+2. Have an Actor with the C++ script to query [Geocoding service](https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm).
+3. The HTTPClient is used to query the Geocoding service.
+4. The [`findAddressCandidates`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm) operation is used for finding a place.
+5. The [`reverseGeocode`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm) operation is used for finding an address. The marker should have the [**ArcGIS Location Component**](https://developers.arcgis.com/unity/maps/location-component/) attached and its values will be used to query the address of the location.
+6. Use the game engine's [raycast](https://docs.unrealengine.com/5.0/en-US/API/Runtime/Engine/Engine/UWorld/LineTraceSingleByChannel/) feature to determine the height at the queried location to account for elevation to place the marker.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:27]:
```suggestion
6. Place a marker using the game engine's [raycast](https://docs.unity3d.com/ScriptReference/Physics.Raycast.html) feature to account for elevation at the queried position.
```

**Review by avr757 — APPROVED** [review]

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 2]
```diff
@@ -1,19 +1,30 @@
-# Geocoding
+# Find places or search for an address
 
-Find the location for an address or find the address of a selected location using the Geocoding service. 
+Find the place for an address or find the address of a selected location using the Geocoding service.
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:3]:
```suggestion
Find the location for an address or find the address of a selected location using the Geocoding service.
```

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
-2. Enter an address in the text field to get the location of the address.
-3. While holding shift, left-click on a location on the map to get the address (or description) of that location.
-4. Raycasts are used to determine the height at the queried location to account for elevation.
-5. This sample uses Esri's [Geocoding service API](https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm), which requires an API Key. You can learn more about [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/) and [Accounts](https://developers.arcgis.com/documentation/mapping-apis-and-services/deployment/accounts/) in the _Mapping APIs and location services_ guide. [Configure your API key](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/tutorials/create-and-manage-an-api-key/#set-service-scopes) to support Geocoding capabilities. 
+1. Open the **Geocoding** level.
+2. Click on the **ArcGISMap Actpr** in the **Outliner**.
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:12]:
```suggestion
2. Click on the **ArcGISMap Actor** in the **Outliner**.
```

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
+3. Set your API key in the **Details** panel.
+4. Click play.
+5. To find a place, put a place name or address in the search bar and press Enter. For information about the type of locations that are supported by the Geocoding service, see the [ArcGIS REST APIs documentation](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm).
+6. To find an address of a place, drop the marker to your point of interest by clicking in the viewport while holding Shift, and you will see the address displayed above the marker.
 
-**Note:** This sample is only set up to work with mouse and keyboard.
+**Note:** This sample is only set up to work with a mouse and keyboard.
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:18]:
```suggestion
**Note:** This sample is only set up to work with mouse and keyboard.
```

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 5]
```diff
@@ -1,19 +1,30 @@
...
+
+## How it works
 
+1. Have a map with the mesh colliders enabled in the level. The mesh collider is required for using raycast. 
+2. Have an Actor with the C++ script to query [Geocoding service](https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm).
+3. The HTTPClient is used to query the Geocoding service.
+4. The [`findAddressCandidates`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm) operation is used for finding a place.
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:25]:
```suggestion
4. The [`findAddressCandidates`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm) API is used for finding a place.
```

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 6]
```diff
@@ -1,19 +1,30 @@
...
+5. The [`reverseGeocode`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm) operation is used for finding an address. The marker should have the [**ArcGIS Location Component**](https://developers.arcgis.com/unity/maps/location-component/) attached and its values will be used to query the address of the location.
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:26]:
```suggestion
5. The [`reverseGeocode`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm) API is used for finding an address. The marker should have the [**ArcGIS Location Component**](https://developers.arcgis.com/unity/maps/location-component/) attached and its values will be used to query the address of the location.
```

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md [hunk 7]
```diff
@@ -1,19 +1,30 @@
...
+6. Use the game engine's [raycast](https://docs.unrealengine.com/5.0/en-US/API/Runtime/Engine/Engine/UWorld/LineTraceSingleByChannel/) feature to determine the height at the queried location to account for elevation to place the marker.
 
 ## Tags
 geocode, reverse geocode, address, locate, search
```
Comment (Jade-JadeH) [inline sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:30]:
```suggestion
geocoding, reverse geocoding, address, locate, search
```

Comment (Jade-JadeH) [reply sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:30]: geocode is the code, geocoding is the process

File: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md (hunk 5 above)
Comment (tTakeda09) [reply sample_project/Content/SampleViewer/Samples/Geocoding/readme.md:25]: This is how it's expressed in the REST API doc so I think it's better to align.
https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm

image

**Pull request merged**